pkg/compactor/bucket_compactor.go

```diff
@@ -877,6 +875,21 @@ func (c *BucketCompactor) Compact(ctx context.Context, maxCompactionTime time.Du
 	return nil
 }
 
+// blockMaxTimeDeltas returns a slice of the difference between now and the MaxTime of each
+// block that will be compacted as part of the provided jobs, in seconds.
+func (c *BucketCompactor) blockMaxTimeDeltas(now time.Time, jobs []*Job) []float64 {
+	var out []float64
+	nowMs := now.UnixMilli()
+
+	for _, j := range jobs {
+		for _, m := range j.Metas() {
+			out = append(out, float64((nowMs-m.MaxTime)/1000))
```

Comment:
nit: I'd suggest more explicit:

```suggestion
			out = append(out, now.Sub(time.UnixMilli(m.MaxTime)).Seconds())
```

and remove `nowMs`.